### PR TITLE
metrics fix

### PR DESCRIPTION
**Pull request by sharadregoti**

### CODE REVIEW

File: gateway/utils/metrics/projects.go [hunk 1]
```diff
@@ -97,7 +98,7 @@ func (m *Module) generateMetricsRequest(project *config.Project, ssl *config.SSL
 func (m *Module) updateSCMetrics(clusterID, projectID string, set, min map[string]interface{}) {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
-	result, err := m.sink.Upsert("config_metrics").Where(types.And(types.Cond("id", "==", clusterID), types.Cond("project", "==", projectID))).Set(set).Min(min).Apply(ctx)
+	result, err := m.sink.Upsert("config_metrics").Where(types.And(types.Cond("cluster_id", "==", clusterID), types.Cond("project", "==", projectID))).Set(set).Min(min).Apply(ctx)
```
Comment (YourTechBud) [inline gateway/utils/metrics/projects.go:101]: Why don't you just put the where clause on the id field?

File: gateway/main.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ var essentialFlags = []cli.Flag{
 		Name:   "cluster",
 		Usage:  "The cluster id to start space-cloud with",
 		EnvVar: "CLUSTER_ID",
-		Value:  "default-cluster",
+		Value:  ksuid.New().String(),
```
Comment (YourTechBud) [inline gateway/main.go:57]: Is this okay? Are you able to specify a clusterid via an env variable? See if you custom id actually gets applied or not?

File: gateway/utils/metrics/projects.go (hunk 1 above)
Comment (sharadregoti) [reply gateway/utils/metrics/projects.go:101]: ok

File: gateway/main.go (hunk 2 above)
Comment (sharadregoti) [reply gateway/main.go:57]: ok, i'll test with env variable

Comment (sharadregoti) [reply gateway/main.go:57]: yes, it works

Comment (YourTechBud) [reply gateway/main.go:57]: Awesome

File: gateway/utils/metrics/projects.go [hunk 3]
```diff
@@ -34,7 +35,8 @@ func (m *Module) generateMetricsRequest(project *config.Project, ssl *config.SSL
 
 	modules := project.Modules
 	set["project"] = project.ID
-
+	id := fmt.Sprintf("%s--%s", clusterID, projectID)
+	set["id"] = id
```
Comment (YourTechBud) [inline gateway/utils/metrics/projects.go:39]: Remove this from set since its already in where

File: gateway/utils/metrics/projects.go [hunk 4]
```diff
@@ -34,7 +35,8 @@ func (m *Module) generateMetricsRequest(project *config.Project, ssl *config.SSL
 
 	modules := project.Modules
 	set["project"] = project.ID
```
Comment (YourTechBud) [inline gateway/utils/metrics/projects.go:37]: Set the clusterid and project where you create the set object

**Pull request merged**